### PR TITLE
BaSP-TG: Added projects endpoints for create, edit and delete.

### PR DESCRIPTION
**Pull request by javierperezraffo**

### CODE REVIEW

**Review by HoracioDuarteOK — APPROVED** [review]

File: src/data/projects.json [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    {
+        "id": "5",
+        "name": "Domainer",
+        "description": "Zoozzy",
+        "startDate": "6/15/2020",
+        "endDate": "7/17/2022",
+        "clientName": "Rosalynd Carnalan"
+    },
+    {
+        "id": "1",
```
Comment (Ignacio-Sormani) [inline src/data/projects.json:43]: Ids should be unique. There are two projects with same id: "1". Also please update the ids so they are a string of 6 digits

File: src/resources/projects.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
+const fs = require('fs');
+const path = require('path');
+const express = require('express');
+
+const projectsFile = path.join(__dirname, '..', 'data', 'projects.json');
+
+const router = express.Router();
+
+router.post('/postProject', (req, res) => {
+  const project = req.body;
```
Comment (Ignacio-Sormani) [inline src/resources/projects.js:10]: Update this method so it creates the id automatically and don't receive it in the body of the endpoint

File: src/resources/projects.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+  let valid = true;
+
+  requiredProps.forEach((prop) => {
+    if (project[prop] === undefined) {
+      valid = false;
+    }
+  });
+
+  if (valid === false) {
```
Comment (Ignacio-Sormani) [inline src/resources/projects.js:29]: Change `valid === false` to `!valid`

File: src/resources/projects.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    return res.sendStatus(400);
```
Comment (Ignacio-Sormani) [inline src/resources/projects.js:30]: Add a message with a description of the error

File: src/resources/projects.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    const storedProjects = JSON.parse(data);
+    storedProjects.push(project);
+    fs.writeFile(projectsFile, JSON.stringify(storedProjects), () => {
+      res.sendStatus(200);
+    });
+  });
+  return true;
+});
+
+router.patch('/editProject/:id', (req, res) => {
```
Comment (Ignacio-Sormani) [inline src/resources/projects.js:43]: I think we should use put, not patch. We should use same method as the rest of the team

File: src/resources/projects.js [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+  const projectId = req.params.id;
+  const updatedProject = req.body;
+
+  fs.readFile(projectsFile, 'utf8', (err, data) => {
+    let storedProjects = JSON.parse(data);
+
+    storedProjects = storedProjects.map((project) => {
+      if (project.id === projectId) {
+        return updatedProject;
```
Comment (Ignacio-Sormani) [inline src/resources/projects.js:52]: The id of the project should be the same of the list, should not be modified by the body of the request. 

```
return { 
   ...updatedProject,
   id: project.id
}
```

File: src/resources/projects.js [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+      res.sendStatus(200);
+    });
+  });
+});
+
+router.delete('/deleteProject/:id', (req, res) => {
+  const projectId = req.params.id;
+
+  if (projectId === undefined) {
+    return res.sendStatus(400);
```
Comment (Ignacio-Sormani) [inline src/resources/projects.js:67]: Add a message with the description of the error

File: src/resources/projects.js [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+  }
+
+  fs.readFile(projectsFile, 'utf8', (err, data) => {
+    const storedProjects = JSON.parse(data);
+    const filtered = storedProjects.filter((project) => project.id !== projectId);
+    fs.writeFile(projectsFile, JSON.stringify(filtered), () => {
+      res.sendStatus(200);
```
Comment (Ignacio-Sormani) [inline src/resources/projects.js:74]: Add a message with the description of what happened

File: src/resources/projects.js [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+
+    storedProjects = storedProjects.map((project) => {
+      if (project.id === projectId) {
+        return updatedProject;
+      }
+      return project;
+    });
+
+    fs.writeFile(projectsFile, JSON.stringify(storedProjects), () => {
+      res.sendStatus(200);
```
Comment (Ignacio-Sormani) [inline src/resources/projects.js:58]: Add a message with the description of what happened

File: src/resources/projects.js [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+  }
+
+  fs.readFile(projectsFile, 'utf8', (err, data) => {
+    const storedProjects = JSON.parse(data);
+    storedProjects.push(project);
+
+    if (err) {
+      res.status(400).send('Cannot save new Admin');
+    } else {
+      res.status(201).send('Admin created');
```
Comment (HoracioDuarteOK) [inline src/resources/projects.js:40]: Error messege

File: src/resources/projects.js [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+  if (!valid) {
+    return res.status(400).send('Cannot create project');
+  }
+
+  fs.readFile(projectsFile, 'utf8', (err, data) => {
+    const storedProjects = JSON.parse(data);
+    storedProjects.push(project);
+
+    if (err) {
+      res.status(400).send('Cannot save new Admin');
```
Comment (HoracioDuarteOK) [inline src/resources/projects.js:38]: error mess

**Review by MatiasPazConti — APPROVED** [review]

**Review by Ignacio-Sormani — APPROVED** [review]

**Pull request merged**